### DISCUSSION TITLE
[Bug]: DefaultValues from useForm is not updated correctly with useFormContext

### DISCUSSION THREAD

**Discussion by zubinshi1980** [react-hook-form/react-hook-form#6140] [Q&A]
### Version Number

7.12.2

### Codesandbox/Expo snack

https://codesandbox.io/s/sad-franklin-d4o5z

### Steps to reproduce

The Default Value set in useForm itself works fine. however, when this used with useFormContext, it is not functional correctly; as it only take the first init. value.

see example: https://codesandbox.io/s/sad-franklin-d4o5z

### Expected behaviour

on the example, when the dialog pop up, the target value changed to "test", i am expecting the value got filled by "test" too. however, it didn't work. it only take the value that init. at the start, when it is changed, it didn't reflect.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
any async `defaultValues` use `reset`

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api/useform/reset

**Reply by zubinshi1980** [reply]
not sure i understand this.  on doc:

// ❌ avoid the following with deep nested default values
const defaultValues = { object: { deepNest: { file: new File() } } };
useForm({ defaultValues });
reset(defaultValues); // share the same reference

// ✅ it's safer with the following, as we only doing shallow clone with defaultValues
useForm({ deepNest: { file: new File() } });
reset({ deepNest: { file: new File() } });

the issue i was encountering is that when we passing the default value (from app.js -> dialog) from parent, it didn't refresh it.  Would you provide example to resolve this if this is not a bug?

thanks

**Reply by zubinshi1980** [reply]
nvm, i got it resolved. need to use reset every single time when the value is changed

**Reply by bluebill1049** [reply]
defaultValues just as the name, it's only for the first mount not the same as local state.
